src/main/java/org/folio/search/configuration/KafkaConfiguration.java

```diff
@@ -45,17 +44,22 @@
     var factory = new ConcurrentKafkaListenerContainerFactory<String, ResourceEventBody>();
     factory.setBatchListener(true);
     factory.setConsumerFactory(jsonNodeConsumerFactory());
-    factory.setBatchErrorHandler(kafkaFactoryErrorHandler());
     return factory;
   }
 
   /**
    * Constructs a batch handler that tries to deliver messages 10 times with configured interval, if exception is not
-   * resolved than messages will be redelivered by next poll() call.
+   * resolved than messages will be redelivered by next poll() call. Creates retry template to consume messages from
+   * kafka.
+   *
+   * @return created {@link RetryTemplate} object
    */
-  private BatchErrorHandler kafkaFactoryErrorHandler() {
-    return new RecoveringBatchErrorHandler(new FixedBackOff(
-      folioKafkaProperties.getRetryIntervalMs(), folioKafkaProperties.getRetryDeliveryAttempts()));
+  @Bean(name = KAFKA_RETRY_TEMPLATE_NAME)
+  public RetryTemplate kafkaMessageListenerRetryTemplate() {
+    return RetryTemplate.builder()
+      .maxAttempts((int) folioKafkaProperties.getRetryDeliveryAttempts())
```

Comment:
Absence of the value in case of casting to (int) would cause NullPointer? Can it be casted to (Integer)?

Reply:
By default - this value is equal to 10 in the FolioKafkaProperties class, just in case, if no configuration has been provided.